src/transformers/pipelines/image_segmentation.py

```diff
@@ -0,0 +1,154 @@
+
+        Args:
+            images (:obj:`str`, :obj:`List[str]`, :obj:`PIL.Image` or :obj:`List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing an HTTP(S) link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images. Images in a batch must all be in the
+                same format: all as HTTP(S) links, all as local paths, or all as PIL images.
+            threshold (:obj:`float`, `optional`, defaults to 0.9):
+                The probability necessary to make a prediction.
+            subtask (:obj:`str`, `optional`, defaults to None):
+                Type of image segmentation: a value from set (semantic, panoptic, part, instance, part-aware)
+
+        Return:
+            A dictionary or a list of dictionaries containing the result. If the input is a single image, will return a
+            dictionary, if the input is a list of several images, will return a list of dictionaries corresponding to
+            each image.
+
+            The dictionaries contain the following keys:
+
+            - **png_string** (:obj:`str`) -- base64 string of a PNG image that contain masks information. Per-pixel
+              segment ids are stored as the PNG image. Each segment is assigned a unique id. Unlabeled pixels (void)
+              are assigned a value of 0. Note that when you load the PNG as an RGB image, you will need to compute the
+              ids via ids=R+G*256+B*256^2 (inspired by https://cocodataset.org/#format-data).
+            - **segments_info** (:obj:`List[dict]`) -- list that contains labels for segments.
+
+            **segments_info** contains dictionaries with the following keys:
+
+            - **id** (:obj:`int`) -- The id attributed for segment.
+            - **label** (:obj:`str`) -- The class label identified by the model.
+            - **score** (:obj:`float`) -- The score attributed by the model for that label.
+        """
+
+        return super().__call__(*args, **kwargs)
+
+    def preprocess(self, image):
+        image = self.load_image(image)
+        target_size = torch.IntTensor([[image.height, image.width]])
+        inputs = self.feature_extractor(images=[image], return_tensors="pt")
+        inputs["target_size"] = target_size
+        return inputs
+
+    def _forward(self, model_inputs):
+        target_size = model_inputs.pop("target_size")
+        outputs = self.model(**model_inputs)
+        model_outputs = {"outputs": outputs, "target_size": target_size}
+        return model_outputs
+
+    def postprocess(self, model_outputs, threshold=0.9, subtask=None):
+        raw_annotations = self.feature_extractor.post_process_segmentation(
+            model_outputs["outputs"], model_outputs["target_size"], threshold=threshold, subtask=subtask
```

Comment:
`subtask` will be generic here and work for ALL models, meaning ALL `feature_extractor` for images that support segmentation needs to accept the `subtask` kwarg (and all its variants).

I am not sure we desire this. Can't we just use `part-aware` all the time for `detr` as it's the most general (If I understand correctly) ?

Other option would be to ask the model it's fullest possible description and do something within the pipeline to actually recover a common representation (`token-classification` 's `aggregate_strategy` is an example of what I mean)

Looking at the proposed changes for masks, I am pretty sure we can drop `subtask` altogether.